include/conf/load.hpp

```diff
@@ -37,17 +38,60 @@ namespace color_coded
       }
     }
 
-    inline args_t load(std::string const &file, std::string const &filetype)
+    inline args_t load_compilation_database(std::string const &file, fs::path filename)
     {
-      if(file.empty())
-      { return defaults(filetype); }
+      static const std::string source_extensions[] {".c", ".cpp", ".cc"};
+      static const std::string header_extensions[] {".h", ".hpp", ".hh"};
+      std::string error;
+      auto const database_ptr(::clang::tooling::JSONCompilationDatabase::loadFromFile(file, error));
+      if(!database_ptr)
+      { return {}; }
```

Comment:
Do we want to echo the error here back to vim? Or is it something we should ignore?